auto-self-nominate/index.ts

```diff
@@ -0,0 +1,169 @@
+                    ),
+                );
+            })
+            .catch(reject);
+    });
+}
+
+async function sendSelfNominateTransaction(
+    accountInfo: { account: string; passphrase: string },
+    params: { deposit: U64; metadata: string },
+) {
+    const tx = createSelfNomination(params);
+    const { account, passphrase } = accountInfo;
+    await sdk.rpc.account.sendTransaction({
+        tx,
+        account,
+        passphrase,
+    });
+}
+
+function createSelfNomination(params: { deposit: U64; metadata: string }): Custom {
+    const { deposit, metadata } = params;
+    const handlerId = new U64(STAKE_ACTION_HANDLER_ID);
+    const actionTag = ACTION_TAG_SELF_NOMINATE;
+    const bytes = RLP.encode([actionTag, deposit, metadata]);
+    return new Custom(
+        {
+            handlerId,
+            bytes,
+        },
+        networkId,
+    );
+}
+
+async function getCandidatesInfo(bestBlockNumber: number): Promise<CandidatesInfo> {
+    const retval: CandidatesInfo = {};
+    const data = (await sdk.rpc.engine.getCustomActionData(2, ["Candidates"], bestBlockNumber))!;
+    const list: Buffer[][] = RLP.decode(Buffer.from(data, "hex"));
+    list.forEach(
+        ([encodedPubkey, encodedDeposit, encodedNominationEnd]) =>
+            (retval[decodePlatformAddressFromPubkey(encodedPubkey).value] = [
+                decodeNumber(encodedNominationEnd),
+                decodeNumber(encodedDeposit),
+            ]),
+    );
+    return retval;
+}
+
+async function needsNomination(
+    info: CandidatesInfo,
+    bestBlockNumber: number,
+    address: string,
+): Promise<boolean> {
+    if (!info.hasOwnProperty(address)) {
+        throw new Error(`SelfNominate first with specific deposit vakue before repeating`);
+    }
+    const [, nominationEndAt] = info[address];
+    const currentTermId = (await getCurrentTermId(bestBlockNumber))!;
+    return nominationEndAt - currentTermId < NEED_NOMINATION_UNDER_TERM_LEFT;
+}
+
+function supplementaryDeposit(
+    info: CandidatesInfo,
+    address: string,
+    targetDeposit: number,
+): number {
+    if (!info.hasOwnProperty(address)) {
+        throw new Error(`SelfNominate first with specific deposit vakue before repeating`);
+    }
+    const [deposit] = info[address];
+    if (deposit < targetDeposit) {
+        return targetDeposit - deposit;
+    } else {
+        return 0;
+    }
+}
+
+async function main() {
+    const accountAddress = getConfig("ACCOUNT_ADDRESS");
+    const passphrase = getConfig("PASSPHRASE");
+    const metadata = getConfig("METADATA");
+    const targetDeposit = parseInt(getConfig("TARGET_DEPOSIT"), 10);
+
+    setInterval(async () => {
+        const bestBlockNumber = await sdk.rpc.chain.getBestBlockNumber()!;
+        const info = await getCandidatesInfo(bestBlockNumber);
+        if (needsNomination(info, bestBlockNumber, accountAddress)) {
```

Comment:
```js
if (await needsNomi...
```

Reply:
How about setting https://palantir.github.io/tslint/rules/strict-boolean-expressions/?

Reply:
Fixed and set the rule `strict-boolean-expression` true.